vpr/src/place/annealer.cpp

```diff
@@ -0,0 +1,898 @@
+        create_move_outcome = move_generator.propose_move(blocks_affected_, proposed_action, rlim, placer_opts_, criticalities_);
+    }
+
+    if (proposed_action.logical_blk_type_index != -1) { //if the agent proposed the block type, then collect the block type stat
+        ++move_type_stats_.blk_type_moves[proposed_action.logical_blk_type_index][(int)proposed_action.move_type];
+    }
+
+    if constexpr (VTR_ENABLE_DEBUG_LOGGING_CONST_EXPR) LOG_MOVE_STATS_PROPOSED();
+
+    VTR_LOGV_DEBUG(g_vpr_ctx.placement().f_placer_debug,
+                   "\t\tBefore move Place cost %e, bb_cost %e, timing cost %e\n",
+                   costs_.cost, costs_.bb_cost, costs_.timing_cost);
+
+    e_move_result move_outcome = e_move_result::ABORTED;
+
+    if (create_move_outcome == e_create_move::ABORT) {
+        if constexpr (VTR_ENABLE_DEBUG_LOGGING_CONST_EXPR) {
+            LOG_MOVE_STATS_OUTCOME(std::numeric_limits<double>::quiet_NaN(),
+                                   std::numeric_limits<double>::quiet_NaN(),
+                                   std::numeric_limits<double>::quiet_NaN(), "ABORTED",
+                                   "illegal move");
+        }
+
+        move_outcome = e_move_result::ABORTED;
+
+    } else {
+        VTR_ASSERT(create_move_outcome == e_create_move::VALID);
+
+        /*
+         * To make evaluating the move simpler (e.g. calculating changed bounding box),
+         * we first move the blocks to their new locations (apply the move to
+         * blk_loc_registry.block_locs) and then compute the change in cost. If the move
+         * is accepted, the inverse look-up in place_ctx.grid_blocks is updated
+         * (committing the move). If the move is rejected, the blocks are returned to
+         * their original positions (reverting blk_loc_registry.block_locs to its original state).
+         *
+         * Note that the inverse look-up place_ctx.grid_blocks is only updated after
+         * move acceptance is determined, so it should not be used when evaluating a move.
+         */
+
+        /* Update the block positions */
+        blk_loc_registry.apply_move_blocks(blocks_affected_);
+
+        //Find all the nets affected by this swap and update the wiring costs.
+        //This cost value doesn't depend on the timing info.
+        //
+        //Also find all the pins affected by the swap, and calculates new connection
+        //delays and timing costs and store them in proposed_* data structures.
+        net_cost_handler_.find_affected_nets_and_update_costs(delay_model_, criticalities_, blocks_affected_,
+                                                              bb_delta_c, timing_delta_c);
+
+        //For setup slack analysis, we first do a timing analysis to get the newest
+        //slack values resulted from the proposed block moves. If the move turns out
+        //to be accepted, we keep the updated slack values and commit the block moves.
```

Comment:
This comment should be moved inside the if, as it is only for the SLACK_TIMING_PLACE option.  Also would be good to say this option evaluates moves with up-to-date timing information, which is more expensive but more accurate.